pkg/controller/controlplane/valuesprovider.go

```diff
@@ -303,13 +333,79 @@ func isOverlayEnabled(networking *gardencorev1beta1.Networking) (bool, error) {
 }
 
 // getControlPlaneShootChartValues collects and returns the control plane shoot chart values.
-func (vp *valuesProvider) getControlPlaneShootChartValues(cluster *extensionscontroller.Cluster) (map[string]interface{}, error) {
+func (vp *valuesProvider) getControlPlaneShootChartValues(cluster *extensionscontroller.Cluster, cp *apismetal.ControlPlaneConfig) (map[string]interface{}, error) {
 	if cluster.Shoot == nil {
 		return nil, fmt.Errorf("cluster %s does not contain a shoot object", cluster.ObjectMeta.Name)
 	}
 
+	metallb, err := getMetallbChartValues(cp)
+	if err != nil {
+		return nil, err
+	}
 	return map[string]interface{}{
 		metal.CloudControllerManagerName: map[string]interface{}{"enabled": true},
+		metal.MetallbName:                metallb,
 	}, nil
+}
 
+// getMetallbChartValues collects and returns the CCM chart values.
+func getMetallbChartValues(
+	cpConfig *apismetal.ControlPlaneConfig,
+) (map[string]interface{}, error) {
```

Comment:
`map[string]any` is more concise than `map[string]interface{}`. That replacement can be done in multiple places. (It's a stylistic change though, that could be done on the whole repo later.)

Reply:
I have replaced the occurrences in whole repo